src/flint/fmpq_series.pyx

```diff
@@ -62,6 +69,23 @@ cdef class fmpq_series(flint_series):
                 raise ZeroDivisionError("cannot create fmpq_series with zero denominator")
             fmpq_poly_scalar_div_fmpz(self.val, self.val, (<fmpz>den).val)
 
+    def __richcmp__(s, t, int op):
+        cdef bint r
+        if op != 2 and op != 3:
+            raise TypeError("series cannot be ordered")
+        s = any_as_fmpq_series(s)
+        if t is NotImplemented:
+            return s
+        t = any_as_fmpq_series(t)
+        if t is NotImplemented:
+            return t
+        r = fmpq_poly_equal((<fmpq_series>s).val, (<fmpq_series>t).val)
+        if r:
+            r = (<fmpq_series>s).prec == (<fmpq_series>t).prec
```

Comment:
Is this the kind of equality comparison we want for series?

- A series cannot compare equal to anything that is not a series e.g. `fmpq_series([1]) != 1`
- An `fmpz_series` can compare equal to an `fmpq_series`: `fmpq_series([1]) == fmpz_series([1])`
- Two series with different prec compare unequal: `fmpz_series([1],5) != fmpz_series([1],6)`
- Invalid series compare equal: `fmpz_series([],-1) == fmpz_series([],-1)`